platform/darwin/src/MGLOfflineStorage.h

```diff
@@ -132,6 +133,30 @@ typedef void (^MGLOfflinePackAdditionCompletionHandler)(MGLOfflinePack * _Nullab
 typedef void (^MGLOfflinePackRemovalCompletionHandler)(NSError * _Nullable error);
 
 /**
+ The type of resource that is requested.
+ */
+typedef NS_ENUM(NSUInteger, MGLResourceKind) {
+    /** Unknown type */
+    MGLResourceKindUnknown,
+    /** Style sheet JSON file */
+    MGLResourceKindStyle,
+    /** TileJSON file as specified in https://www.mapbox.com/mapbox-gl-js/style-spec/#root-sources */
```

Comment:
I suppose we could alternatively refer to the TileJSON file as the file specified by the `configurationURL` parameter in MGLTileSource's initializers.